src/rgw/rgw_op.cc

```diff
@@ -6392,9 +6397,12 @@ int RGWDeleteMultiObj::verify_permission()
         }
       }
     }
+
+    bool empty = rgw::sal::RGWObject::empty(s->object.get()) || s->object->get_instance().empty();
```

Comment:
@dang is this typical usage?

Reply:
I don't think this is right.  It's technically valid, since empty() checks that the name is empty but not the instance, but it should be valid to have an un-instanced object, and this check will consider that to be an empty object.

If this check is as intended, can we change the name of the variable from "empty" to something more descriptive, since an object with a name but no instance is definitely not empty.

Reply:
The thing I'm trying to solve here is that while testing, I found that for multi-object delete, `s->object.get()` was returning null. The other side of the `||` is just what was there already, so I assume it has a purpose but I'm not very familiar with this code. github.com/ceph/ceph/pull/36583/ has a very similar change, but calls it `is_versioned`. Would that be more appropriate? (I generally don't like putting in changes that are already in a separate PR, but I needed this part of the fix to test that the core change was working).

Reply:
Okay, now I'm understanding.  Let's not call the variable "empty".  "is_versioned" would work, but you'll have to negate it.  Maybe "not_versioned" would be simplest?

Reply:
(ah, getting it now)

Reply:
Thanks. Changed to not_versioned.